moveit_core/smoothing_plugins/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+# Plugin implementations
+
+add_library(moveit_butterworth_filter SHARED
```

Comment:
Plugin base and implementation should be separate libraries or even packages (see `kinematics_base` for instance). Otherwise, you would have to depend on the ButterworthFilter to implement your own smoothing plugin which doesn't make a lot of sense.

Reply:
I'm fairly sure that's not true. PR #682 adds another smoothing plugin that certainly does not depend on ButterworthFilter. See ruckig_filter.h [here](https://github.com/ros-planning/moveit2/pull/682/files#diff-36de96511bdf77e9c0095e1a897b4b6f2b56dd1a1ce7bfbeb996ccf073e68d47).

Reply:
Also, if you read the [ROS2 plugin tutorial ](https://docs.ros.org/en/foxy/Tutorials/Pluginlib.html)carefully, you'll see that the base class (RegularPolygon) is only a header file. It's not compiled to a library.